core/threaded/scheduler_LET.c

```diff
@@ -456,29 +487,92 @@ void lf_sched_trigger_reaction(reaction_t* reaction, int worker_number) {
     _lf_sched_insert_reaction(reaction);
 }
 
+#ifdef MODAL_REACTORS
+/**
+ * @brief This function takes a self pointer and locks, recursively
+ *  all the parent reactors which are modal. It effectively walks
+ *  up the containment hierarchy and locks any modal reactor
+ * 
+ * @param reactor 
+ */
+static void _lf_sched_lock_modal_parents(self_base_t* reactor, int worker_number) {
+    self_base_t *parent = reactor->parent;
+    // If has a parent/containing reactor
+    if (parent) {
+        // Check if it is modal
+        if (parent->_lf__mode_state.current_mode) { // FIXME: Is this a reliable way to check if reactor is modal?
```

Comment:
@a-sr Is there a clearer way of figuring out if a Reactor is modal or not?

Reply:
This looks like a reasonable way to me. An alternative is to check whether ASTUtils.allModes(reactor) returns an empty list, but I don't think you need to change this.

Reply:
I agree, this is a good way to check for a modal reactor.